### PR TITLE
compute top people locally (WEBAPP-3665)

### PR DESCRIPTION
**Pull request by herrmannplatz**

### CODE REVIEW

File: test/unit_tests/conversation/ConversationServiceSpec.coffee [hunk 1]
```diff
@@ -329,3 +329,31 @@ describe 'z.conversation.ConversationService', ->
...
+        {"conversation":"34e7f58e-b834-4d84-b628-b89b295d46c1","id":"f7adaa16-38f5-483e-b621-72ff1dbd2275","from":"5598f954-674f-4a34-ad47-9e5ee8f00bcd","time":"#{new Date(Date.now() - 4).toISOString()}","data":{"content":"hello","nonce":"f7adaa16-38f5-483e-b621-72ff1dbd2275","previews":[]},"type":"conversation.message-add","category": 16}
+        {"conversation":"34e7f58e-b834-4d84-b628-b89b295d46c2","id":"f7adaa16-38f5-483e-b621-72ff1dbd2275","from":"5598f954-674f-4a34-ad47-9e5ee8f00bcd","time":"#{new Date(Date.now() - 5).toISOString()}","data":{"content":"hello","nonce":"f7adaa16-38f5-483e-b621-72ff1dbd2275","previews":[]},"type":"conversation.message-add","category": 16}
+        {"conversation":"34e7f58e-b834-4d84-b628-b89b295d46c3","id":"f7adaa16-38f5-483e-b621-72ff1dbd2275","from":"5598f954-674f-4a34-ad47-9e5ee8f00bcd","time":"2016-01-09T13:11:15.051Z","data":{"content":"hello","nonce":"f7adaa16-38f5-483e-b621-72ff1dbd2275","previews":[]},"type":"conversation.message-add","category": 16}
```
Comment (lipis) [inline test/unit_tests/conversation/ConversationServiceSpec.coffee]: Let's either remove the space from `"category": 16` or add everywhere else after a delimiter :)

File: app/script/conversation/ConversationRepository.coffee [hunk 2]
```diff
@@ -397,6 +397,20 @@ class z.conversation.ConversationRepository
     return @conversations_unarchived()?[0]
 
   ###
+  Return all list of sorted conversation ids based on the number of messages in the last 30 days.
```
Comment (bennycode) [inline app/script/conversation/ConversationRepository.coffee]: "a list"

File: app/script/conversation/ConversationService.coffee [hunk 3]
```diff
@@ -235,6 +235,28 @@ class z.conversation.ConversationService
       throw error
 
   ###
+  Get recent conversations.
```
Comment (bennycode) [inline app/script/conversation/ConversationService.coffee]: No need for doc?

File: app/script/conversation/ConversationService.coffee [hunk 4]
```diff
@@ -235,6 +235,28 @@ class z.conversation.ConversationService
...
+  get_active_conversations_from_db: ->
+    min_date = new Date()
+    min_date.setDate min_date.getDate() - 30
+
+    @storage_service.db[@storage_service.OBJECT_STORE_CONVERSATION_EVENTS]
+    .where 'time'
+    .between min_date.toISOString(), new Date().toISOString()
+    .toArray()
+    .then (events) ->
+      conversations = events.reduce (acc, event) ->
```
Comment (bennycode) [inline app/script/conversation/ConversationService.coffee]: what is `acc`?

Comment (herrmannplatz) [reply app/script/conversation/ConversationService.coffee]: accumulated

**Pull request merged**